sqld/src/replication/replica/injector.rs

```diff
@@ -143,18 +22,25 @@ impl<'a> FrameInjector<'a> {
             hook_ctx,
         )?;
 
-        Ok(Self { conn, ctx })
+        Ok(Self { conn })
     }
 
-    /// sets the injector's frames to the provided frames, trigger a dummy write, and collect the
-    /// injection result.
-    fn inject_frames(&mut self, frames: Frames) -> anyhow::Result<FrameNo> {
-        self.ctx.set_frames(frames);
-
-        let _ = self
-            .conn
-            .execute("create table if not exists __dummy__ (dummy)", ());
-
-        self.ctx.take_result()
+    pub fn step(&mut self) -> anyhow::Result<bool> {
+        let res = self.conn.execute("create table __dummy__ (dummy);", ());
+
+        match res {
+            Ok(_) => panic!("replication hook was not called"),
```

Comment:
ok, so `SQLITE_EXIT_REPLICATION` is the expected way to signal successful replication? Up above, I saw that returning `SQLITE_EXIT_REPLICATION` also triggered a `tracing::warn!` message, so I suppose it should be degraded to debug/trace.

side note: I'm also not a huge fan of using error codes for success paths, but that's a preexisting condition with SQLite and its SQLITE_ROW, SQLITE_BUSY and friends, so never mind, just a random rant (:

Reply:
So the reason why we don't simply use SQLITE_OK because sqlite will assume that the dummy statement was successful and cache a schema that the `__dummy__` table exists (even though it doesn't because we never applied the frames!).